### PR TITLE
feat: #191 Show user name

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 1]
```diff
@@ -12,12 +14,29 @@ import styles from "src/logic/userPage/UserPage.module.scss";
  */
 export const UserPage = () => {
   const {uuid} = useParams<QueryParamTypes>();
+  const [userName, setUserName] = useState<string>("");
+
+  if (!uuid) {
+    throw new Error("User is not exist");
```
Comment (Shustovsky) [inline mw-webapp/src/logic/userPage/UserPage.tsx:20]: maybe if the user has not logged in, it should be redirected to another page (main, registration or 404) ?

Comment (Howl404) [reply mw-webapp/src/logic/userPage/UserPage.tsx:20]: You are mistaking it for UserProfile, this is UserPage (like a WayPage) we are getting it by Uuid in url. But i think we should check if user with provided Uuid exist inside getUser function

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 1 above)
Comment (tritonJS826) [reply mw-webapp/src/logic/userPage/UserPage.tsx:20]: 1. Uuid should be described near other page parameters and should be validated in some page wrapper. or at least hook.
2. If it is undefined or just user with the same name does not exist, user should be redirected to 404

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 2]
```diff
@@ -12,12 +14,29 @@ import styles from "src/logic/userPage/UserPage.module.scss";
...
+  }
+
+  /**
+   * Set user name
```
Comment (tritonJS826) [inline mw-webapp/src/logic/userPage/UserPage.tsx:24]: So set or get? Let's use word "load". loadUser

**Review by tritonJS826 — APPROVED** [review]
Good job)

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx]: Moved check uuid to Routes

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx]: Changed

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 3]
```diff
@@ -1,52 +1,64 @@
...
 import {FavoriteWaysTable} from "src/logic/waysTable/FavoriteWaysTable";
 import {MentoringWaysTable} from "src/logic/waysTable/MentoringWaysTable";
 import {OwnWaysTable} from "src/logic/waysTable/OwnWaysTable";
-import {pages} from "src/router/pages";
 import styles from "src/logic/userPage/UserPage.module.scss";
 
+/**
+ * PageProps
+ */
+interface PageProps {
```
Comment (tritonJS826) [inline mw-webapp/src/logic/userPage/UserPage.tsx:13]: -> UserPageProps

File: mw-webapp/src/router/Router.tsx [hunk 4]
```diff
@@ -1,10 +1,14 @@
-import {Route, Routes} from "react-router-dom";
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
```
Comment (tritonJS826) [inline mw-webapp/src/router/Router.tsx:1]: why?

File: mw-webapp/src/router/pages.tsx [hunk 5]
```diff
@@ -21,7 +21,11 @@ export const pages = {
      * @returns {string} path to react-router OR Specific URl path to specific user
      */
     path: (uuid: string): string => `/user/${uuid}`,
-    element: <UserPage />,
+
+    /**
+     * Sdf
```
Comment (tritonJS826) [inline mw-webapp/src/router/pages.tsx:26]: ?

File: mw-webapp/src/router/pages.tsx [hunk 6]
```diff
@@ -21,7 +21,11 @@ export const pages = {
...
+     */
+    element: (uuid: string) => <UserPage uuid={uuid} />,
```
Comment (tritonJS826) [inline mw-webapp/src/router/pages.tsx:28]: looks like page itself, element is super absract thing, be more business specific 
 element -> pageComponent

File: mw-webapp/src/router/useValidatedParams.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+   * Page's uuid
+   */
+  uuid: string;
+}
+
+/**
+ * If query params exist allow redirect to Page and if not navigate to Page404
+ */
+export const ValidatedParams = (props: PropsWithChildren) => {
+  const {uuid} = useParams<QueryParamTypes>();
```
Comment (tritonJS826) [inline mw-webapp/src/router/useValidatedParams.tsx:20]: does every page shpould has validate params? Good prototype, but it does not look reusable

Comment (Ekaterina1994) [reply mw-webapp/src/router/useValidatedParams.tsx:20]: Changed validated wrapper but also dont know how to make it reusable(((

**Review by tritonJS826 — APPROVED** [review]
yeap, good enough!

**Review by Shustovsky — APPROVED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx [hunk 8]
```diff
@@ -1,52 +1,63 @@
...
-  const {uuid} = useParams<QueryParamTypes>();
+export const UserPage = (props: UserPageProps) => {
+  const [userName, setUserName] = useState<string>("");
+
+  /**
+   * Load user
+   */
+  const getUser = async () => {
+    const user = await UserPreviewDAL.getUserPreview(props.uuid);
+    setUserName(user.name);
```
Comment (Howl404) [inline mw-webapp/src/logic/userPage/UserPage.tsx:31]: Please do something if user is not found

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/logic/userPage/UserPage.tsx (hunk 8 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/userPage/UserPage.tsx:31]: We created ValidateParams wrapper for check params (uuid and others). Or maybe I don't understand you?

**Review by Howl404 — APPROVED** [review]

**Pull request merged**